### PR TITLE
Feat/2.1 angular guide

### PR DESCRIPTION
**Pull request by iam4x**
This PR addresses the #2214 issue 👍 

Now we "officially support" Angular2+ integrations via the connector API. It's a first step in the direction of a angular-instantsearch library.

Need your last review on typos or anything else.

### CODE REVIEW

File: docgen/src/guides/angular-integration.md [hunk 1]
```diff
@@ -0,0 +1,366 @@
+---
+title: Integrate with Angular2+
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: Integrate with Angular (2+)

File: docgen/src/guides/angular-integration.md [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+name: integrate-angular
+withHeadings: true
+navHeight: 20
+editable: true
+githubSource: docgen/src/guides/angular-integration.md
+---
+
+## Introduction
+
+In this guide, we will learn together step by step how to integrate InstantSearch.js into an Angular2+ application.
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: In this guide we

Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: Angular (2+)

File: docgen/src/guides/angular-integration.md [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+At the end of this guide you will have an Angular2+ application running with an InstantSearch.js search experience!
+
+For the rest of the guide we assume that you have some Angular2 knowledge and the angular-cli installed on your machine. If it's not the case have a look at the awesome lessons on [egghead.io](https://egghead.io/lessons/angular-2-say-hello-world-to-angular-2).
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: Angular

File: docgen/src/guides/angular-integration.md [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+navHeight: 20
+editable: true
+githubSource: docgen/src/guides/angular-integration.md
+---
+
+## Introduction
+
+In this guide, we will learn together step by step how to integrate InstantSearch.js into an Angular2+ application.
+
+At the end of this guide you will have an Angular2+ application running with an InstantSearch.js search experience!
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: Angular

File: docgen/src/guides/angular-integration.md [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+
+For the rest of the guide we assume that you have some Angular2 knowledge and the angular-cli installed on your machine. If it's not the case have a look at the awesome lessons on [egghead.io](https://egghead.io/lessons/angular-2-say-hello-world-to-angular-2).
+
+If you are new to InstantSearch.js, best is to read first the [getting started](https://community.algolia.com/instantsearch.js/v2/getting-started.html) guide.
+
+Before going through this guide, it's well recommended to read the [customize widgets](https://community.algolia.com/instantsearch.js/v2/guides/customization.html) and the [introduce to connectors](https://community.algolia.com/instantsearch.js/v2/connectors.html) guides for InstantSearch.js
+
+## Create an Angular app
+
+We will create a default Angular application using the angular-cli, this is done like this:
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: `[angular-cli](http://yarn.fyi/@angular/cli)`

File: docgen/src/guides/angular-integration.md [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+
+First of all, you need to add InstantSearch.js to your dependencies:
+
+```sh
+$ yarn add instantsearch.js
+// OR
+$ npm install instantsearch.js --save
+```
+
+After you have installed InstantSearch.js to your application dependencies, we need to instantiate the InstantSearch.js library into an Angular service. So we will generate this new service:
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: application's dependencies

File: docgen/src/guides/angular-integration.md [hunk 7]
```diff
@@ -0,0 +1,366 @@
...
+
+```sh
+$ ng generate service services/instantsearch
+// OR
+$ ng g s services/instantsearch
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: I wouldn't write the shortcuts here.

File: docgen/src/guides/angular-integration.md [hunk 8]
```diff
@@ -0,0 +1,366 @@
...
+```javascript
+import { Injectable } from '@angular/core';
+import instantsearch from 'instantsearch.js/es';
+
+@Injectable()
+export class InstantSearchService {
+  search = instantsearch({
+    appId: 'latency',
+    apiKey: '3d9875e51fbd20c7754e65422f7ce5e1',
+    indexName: 'bestbuy',
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: use the instant_search demo too maybe?

File: docgen/src/guides/angular-integration.md [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+    urlSync: true
+  });
+
+  constructor() {}
+}
+```
+
+`appId`, `apiKey` and `indexName` are mandatory. Those values are credentials of your application in Algolia. They can be found in your [Algolia dashboard](https://www.algolia.com/api-keys).
+
+You can see the code example above we are importing the ES6 module build of the InstantSearch.js library. This will reduce a lot the final build size of your application (this is known as [tree shaking](https://webpack.js.org/guides/tree-shaking/) in the JavaScript world).
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: This will reduce the final build size of your application a lot (

File: docgen/src/guides/angular-integration.md [hunk 10]
```diff
@@ -0,0 +1,366 @@
...
+
+Don't forget to register your service into the providers of your `src/app.modules.ts` file to be able to inject it into your next components. [(Official Angular documentation)](https://angular.io/tutorial/toh-pt4#inject-the-heroservice).
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: This happens automatically when generating, no?

File: docgen/src/guides/angular-integration.md [hunk 11]
```diff
@@ -0,0 +1,366 @@
...
+
+## Create a Hits component
+
+In order to display results, we are going to use the [connectors](https://community.algolia.com/instantsearch.js/v2/connectors.html) API of InstantSearch.js and more specially the [connectHits](https://community.algolia.com/instantsearch.js/v2/connectors/connectHits.html) connector.
+
+Start by creating a new component in your Angular application for the hits:
+
+```sh
+$ ng generate component components/hits -is -it
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: what does `-is -it` actually mean?

**Comment by redox** [comment]
Looks awesome!

File: docgen/src/guides/angular-integration.md [hunk 12]
```diff
@@ -0,0 +1,366 @@
...
+        this.state = state;
+      });
+    }
+
+    this.state = state;
+  }
+}
+```
+
+Now we have a component which is synced to search state. This state search is accessible in the template of the componet.
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: component

File: docgen/src/guides/angular-integration.md [hunk 13]
```diff
@@ -0,0 +1,366 @@
...
+And voilà! You should now see in your app the default search results without query provieded!
+
+This is pretty cool, but it's not usable without a search box, let's build a search box in a similar way.
+
+## Interact with a search box
+
+Create a new component for the SearchBox:
+
+```sh
+$ ng g c components/search-box --it --is
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: in full

File: docgen/src/guides/angular-integration.md [hunk 14]
```diff
@@ -0,0 +1,366 @@
...
+Mount your search box component in your app and you are good to go!
+
+## Refining even further
+
+When you’re searching through data, not everything is easy to know about. That’s why we offer the RefinementList. Implementing a RefinementList is very similar to a SearchBox, but now we have to pass options to the widget.
+
+So just like the other components, we will start with the bootstrapping command:
+
+```sh
+$ ng g c components/refinements-list -it -is
```
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: `refinement-list`

Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: lol that character `ng g c components/refinement-list -it -is` is what I meant

**Comment by Haroenv** [comment]
Netlify build fails on `yarn`, super odd: 

```
[...]
3:39:40 PM: Installing NPM modules using Yarn version 0.18.1
3:39:41 PM: yarn install v0.18.1
3:39:41 PM: [1/4] Resolving packages...
3:39:42 PM: [2/4] Fetching packages...
3:39:46 PM: error An unexpected error occurred: "https://registry.yarnpkg.com/webpack-dev-server/-/webpack-dev-server-2.7.0.tgz: Request failed \"404 Not Found\"".
3:39:46 PM: info If you think this is a bug, please open a bug report with the information provided in "/opt/build/repo/yarn-error.log".
3:39:46 PM: info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
3:39:46 PM: Error during Yarn install
3:39:46 PM: Build complete: exit code: 1
3:39:46 PM: Cleaning up docker container
3:39:46 PM: Error running command: Build script returned non-zero exit code: 1
3:39:46 PM: An error occurred while building the site, skipping the deploy and cache refresh.
3:39:47 PM: Build script returned non-zero exit code: 1
3:39:48 PM: Finished processing build request in 1m7.384102076s
```

**Comment by Haroenv** [comment]
@iam4x You'll have to run `yarn` again, because https://registry.yarnpkg.com/webpack-dev-server/-/webpack-dev-server-2.7.0.tgz is deleted, while https://registry.yarnpkg.com/webpack-dev-server/-/webpack-dev-server-2.7.1.tgz exists (it was probably unpublished within 24 hours of publish)

File: docgen/src/guides/angular-integration.md (hunk 8 above)
Comment (iam4x) [inline docgen/src/guides/angular-integration.md]: Hmm we need to update the attribute name and template of the hits if we do so, this not mandatory I think.

File: docgen/src/guides/angular-integration.md (hunk 10 above)
Comment (iam4x) [inline docgen/src/guides/angular-integration.md]: Nope, that was the issue you went into the first time of the service not accessible.

File: docgen/src/guides/angular-integration.md (hunk 11 above)
Comment (iam4x) [inline docgen/src/guides/angular-integration.md]: --inline-styles --inline-template

**Comment by iam4x** [comment]
Thanks for the review @Haroenv good catch 👍

**Comment by iam4x** [comment]
Build preview of the guide is available here: https://deploy-preview-2285--algolia-instantsearch.netlify.com/guides/angular-integration.html

File: docgen/src/guides/angular-integration.md (hunk 11 above)
Comment (Haroenv) [inline docgen/src/guides/angular-integration.md]: Can you write it in full so it doesn't need explanation?

File: docgen/src/data/communityHeader.json [hunk 15]
```diff
@@ -79,8 +79,11 @@
       {
         "name": "Migrate from V1",
         "url": "guides/migration.html"
+      },
+      {
+        "name": "Integrate with Angular2+",
```
Comment (Haroenv) [inline docgen/src/data/communityHeader.json]: Angular (2+) here too

**Comment by iam4x** [comment]
We should be all good now what do you think @Haroenv ?

File: docgen/layouts/mixins/documentationjs/widget-category-showcase.pug [hunk 16]
```diff
@@ -1,5 +1,5 @@
 mixin widgetCategoryShowCase(category, symbols, title)
-  - const categorySymbols = symbols[category];
+  - const categorySymbols = symbols && symbols[category];
```
Comment (Haroenv) [inline docgen/layouts/mixins/documentationjs/widget-category-showcase.pug]: this is unrelated right?

Comment (iam4x) [inline docgen/layouts/mixins/documentationjs/widget-category-showcase.pug]: Yes, it was just crashing the dev environment.
